ec2/ec2test/availability_zones.go

```diff
@@ -17,15 +17,31 @@ import (
 	"gopkg.in/amz.v3/ec2"
 )
 
-// SetAvailabilityZones initializes the availability zones the test
+// SetAvailabilityZones replaces the availability zones the test
 // server is returning.
+//
+// NOTE: If zones does not contain one or more existing zones those
+// existing zones are removed along with any subnets that are
+// associated with them!
 func (srv *Server) SetAvailabilityZones(zones []ec2.AvailabilityZoneInfo) {
 	srv.mu.Lock()
-	srv.zones = make([]availabilityZone, len(zones))
-	for i, z := range zones {
-		srv.zones[i] = availabilityZone{z}
+	defer srv.mu.Unlock()
+	oldZones := srv.zones
+	srv.zones = make(map[string]availabilityZone)
+	for _, z := range zones {
```

Comment:
What this loop does is:

    iterate over all the new zones removing all subnets associated with each new zone

Did you mean to iterate over oldZones instead?

(In the end, with the current loop, I believe that the only subnets left will be ones associated with zones from oldZones but *not* in new zones. I think the effect you're after is exactly the opposite.)